gpkit/nomials/map.py

```diff
@@ -0,0 +1,181 @@
+"Implements the NomialMap class"
+from collections import defaultdict
+import numpy as np
+from .. import DimensionalityError
+from ..small_classes import HashVector, Quantity, Strings
+from ..small_scripts import mag
+from ..varkey import VarKey
+from .substitution import parse_subs
+
+
+DIMLESS_QUANTITY = Quantity(1, "dimensionless")
+
+
+class NomialMap(HashVector):
+    "Class for efficent algebraic represention of a nomial"
+    units = None
+    expmap = None  # used for monomial-mapping postsubstitution; see .mmap()
+    csmap = None
+
+    def set_units(self, thing, thing2=None):
```

Comment:
As far as I can tell this just got renamed as opposed to splitting into two methods.